python/lsst/ts/MTAOS/Utility.py

```diff
@@ -165,5 +183,41 @@ def getCscName():
     return "MTAOS"
 
 
+def addRotFileHandler(log, filePath, maxBytes=1e6, backupCount=5):
+    """Add a rotating file handler to a logger.
+
+    Note: The input log object will be updated directly.
+
+    Parameters
+    ----------
+    log : logging.Logger
+        Logger.
+    filePath : pathlib.PosixPath
+        File path.
+    maxBytes : int, optional
+        Maximum file size in bytes for each file. (the default is 1e6.)
+    backupCount : int, optional
+        Number of log files to retain. (the default is 5.)
+
+    Returns
+    -------
+    fileHandler : logging.RotatingFileHandler
+        The file handler added.
+    """
+
+    fileHandler = RotatingFileHandler(
+        filename=filePath, mode="a", maxBytes=int(maxBytes),
+        backupCount=int(backupCount))
+
+    logFormat = logging.Formatter(fmt="%(asctime)s - %(levelname)s - %(message)s")
+    fileHandler.setFormatter(logFormat)
+
+    fileHandler.setLevel(logging.DEBUG)
```

Comment:
I am concerned about this for several reasons:
* It conflates two orthogonal logging properties: where the data is written (the log handlers) and what messages are written (the log level). It makes the intention and behavior of the function confusing.
* It is clearer to set the level of the logger. Do you have a compelling reason to have one handler have different verbosity than the other log handler(s) for this logger?

I urge you to remove this line and set the level of the entire log as a separate direct call, e.g. `log.setLevel(logging.DEBUG)`.

Reply:
I checked your code for the `salobj` and realized that the user can use the `Remote` to reset the log level of `log` object. If I need to align the log level of log and file handler, I will need to add the callback function to update the log level of file handler. Therefore, I decide to decouple them and make it simple. And I am shocked that the log level of logger and file handler are decoupled when I checked the Python logging document.
Therefore, just set the default log level of file handler to be `DEBUG` to output everything I need because this is just for the debugging in the tests.

Reply:
I don't understand. If you set the log level of the Logger (self.log) then it should affect all handlers (unless the handler has been set to a specific log level, overriding the default). That is what the setLogLevel command does.

Are you saying you tried that and found that the setLogLevel command did not affect the log level of data logged to the file handler?

Note that a really confusing misfeature of the python logger is that the log level is ignored unless there is at least one handler. Which means that by default setting the log level has no effect. Unfortunately this seems unlikely to ever be fixed, despite the massive confusion this has caused over the years.

Here is the code for the setLogLevel command, minus the doc string:
```
    def do_setLogLevel(self, data):
        self.log.setLevel(data.level)
        self.put_log_level()
```

Anyway, it's not a big deal. I can see arguments for doing it both ways. But if you prefer to keep your code then I beg you to:
* Document that the function sets the log level.
* Make the log level for the file handler an argument, so the user has control.

Reply:
Generate the issue #22 and will resolve it in a latter time.